components/restapi/core/transaction.go

```diff
@@ -5,57 +5,40 @@ import (
 
 	"github.com/iotaledger/hive.go/ierrors"
 	"github.com/iotaledger/inx-app/pkg/httpserver"
-	"github.com/iotaledger/iota-core/pkg/model"
 	iotago "github.com/iotaledger/iota.go/v4"
 	"github.com/iotaledger/iota.go/v4/api"
 )
 
-func blockIDByTransactionID(c echo.Context) (iotago.BlockID, error) {
+func blockIDFromTransactionID(c echo.Context) (iotago.BlockID, error) {
 	txID, err := httpserver.ParseTransactionIDParam(c, api.ParameterTransactionID)
 	if err != nil {
 		return iotago.EmptyBlockID, ierrors.Wrapf(err, "failed to parse transaction ID %s", c.Param(api.ParameterTransactionID))
 	}
 
-	return blockIDFromTransactionID(txID)
+	return deps.RequestHandler.BlockIDFromTransactionID(txID)
 }
 
-func blockIDFromTransactionID(transactionID iotago.TransactionID) (iotago.BlockID, error) {
-	// Get the first output of that transaction (using index 0)
-	outputID := iotago.OutputIDFromTransactionIDAndIndex(transactionID, 0)
-
-	output, spent, err := deps.Protocol.Engines.Main.Get().Ledger.OutputOrSpent(outputID)
-	if err != nil {
-		return iotago.EmptyBlockID, ierrors.Wrapf(echo.ErrInternalServerError, "failed to get output %s: %s", outputID.ToHex(), err)
-	}
-
-	if output != nil {
-		return output.BlockID(), nil
-	}
-
-	return spent.BlockID(), nil
-}
-
-func blockByTransactionID(c echo.Context) (*model.Block, error) {
-	blockID, err := blockIDByTransactionID(c)
+func blockFromTransactionID(c echo.Context) (*iotago.Block, error) {
+	blockID, err := blockIDFromTransactionID(c)
 	if err != nil {
 		return nil, ierrors.Wrapf(echo.ErrBadRequest, "failed to get block ID by transaction ID: %s", err)
 	}
 
-	block, exists := deps.Protocol.Engines.Main.Get().Block(blockID)
-	if !exists {
-		return nil, ierrors.Wrapf(echo.ErrNotFound, "block not found: %s", blockID.ToHex())
+	block, err := deps.RequestHandler.BlockByID(blockID)
+	if err != nil {
+		return nil, ierrors.Wrapf(echo.ErrNotFound, "block not found: %s", err)
```

Comment:
We already added the "block not found" message in `BlockByID`

Reply:
ok, better to simply return the error without wrapping or to add something else as a wrapper?

Reply:
Since we already added the `echo.ErrNotFound` in `BlockByID`, we can simply return the error here without further wrapping.